src/sign/records.rs

```diff
@@ -149,31 +163,34 @@ impl<N, D> SortedRecords<N, D> {
                 }
 
                 // Create and push the RRSIG record.
+                let signature = key.raw_secret_key().sign_raw(&buf).unwrap();
+                let signature = signature.as_ref().to_vec();
+                let Ok(signature) = signature.try_octets_into() else {
+                    return Err(ErrorTypeToBeDetermined);
+                };
```

Comment:
I think that signing an RRset should be a separated function and not be inlined here.